client/gutenberg/editor/edit-post/components/editor-document-head/index.jsx

```diff
@@ -0,0 +1,47 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'components/data/document-head';
+import { getPostType } from 'state/post-types/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export const EditorDocumentHead = ( { isNewPost, postType, postTypeObject, translate } ) => {
+	let title = '';
+	if ( 'post' === postType ) {
+		title = isNewPost ? translate( 'New Post' ) : translate( 'Edit Post' );
+	} else if ( 'page' === postType ) {
+		title = isNewPost ? translate( 'New Page' ) : translate( 'Edit Page' );
+	} else if ( !! postTypeObject ) {
+		title = isNewPost ? postTypeObject.labels.new_item : postTypeObject.labels.edit_item;
+	} else {
+		title = isNewPost ? translate( 'New' ) : translate( 'Edit' );
+	}
```

Comment:
_if_ we split this into a separate function we can more easily test it

```js
export const getTitle = ( { isNewPost, postType, postTypeObject, translate = a => a } ) => {
	if ( 'post' === postType && isNewPost ) {
		return translate( 'New Post' );
	}

	if ( 'post' === postType && ! isNewPost ) {
		return translate( 'Edit Post' );
	}

	if ( 'page' === postType && isNewPost ) {
		return translate( 'New Page' );
	}

	if ( 'page' === postType && ! isNewPost ) {
		return translate( 'Edit Page' );
	}

	if ( !! postTypeObject && isNewPost ) {
		return postTypeObject.labels.new_item;
	}

	if ( !! postTypeObject && ! isNewPost ) {
		return postTypeObject.labels.edit_item;
	}

	if ( isNewPost ) {
		return translate( 'New' );
	}

	return translate( 'Edit' );
}

export const EditorDocumentHead = ( { isNewPost, postType, postTypeObject, translate } ) =>
	<DocumentHead title={ getTitle( { isNewPost, postType, postTypeObject, translate } ) } />
```

Reply:
It looks so cumbersome though.
And it's not like there will be any more cases there.